### PR TITLE
Tidy up validations + bug fixes + deprecations

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Tidy up many modules checks and while at it, fixed a couple of bugs and added a couple of deprecations.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/modules/database/aerospike/aerospike_migrations.py
plugins/modules/monitoring/logstash_plugin.py
plugins/modules/files/iso_extract.py
plugins/modules/identity/opendj/opendj_backendprop.py
plugins/modules/storage/emc/emc_vnx_sg_member.py
plugins/modules/source_control/hg.py
plugins/modules/source_control/bzr.py
plugins/modules/system/runit.py
plugins/modules/storage/ibm/ibm_sa_host_ports.py
plugins/modules/storage/ibm/ibm_sa_vol_map.py
plugins/modules/storage/ibm/ibm_sa_vol.py
plugins/modules/storage/ibm/ibm_sa_pool.py
plugins/modules/storage/ibm/ibm_sa_host.py
plugins/modules/storage/ibm/ibm_sa_host_ports.py
plugins/modules/storage/ibm/ibm_sa_domain.py
plugins/modules/database/vertica/vertica_user.py
plugins/modules/database/vertica/vertica_schema.py
plugins/modules/database/vertica/vertica_role.py
plugins/modules/database/vertica/vertica_info.py
plugins/modules/database/misc/riak.py
plugins/modules/database/misc/kibana_plugin.py
plugins/modules/database/misc/elasticsearch_plugin.py

### CODE REVIEW

**Comment by russoz** [comment]
@felixfontein This is the other "big one" (in terms of # of lines in the ignore files). This one will most definitely require some rebasing when those other two get merged, but they should be quite straightforward: adjacent lines being removed in `ignore*.txt` files only.

**Comment by felixfontein** [comment]
This also needs a rebase.

File: changelogs/fragments/1830-valmod_docmissingtype_batch1.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+bugfixes:
+  - kibana_plugin - ``state`` parameter choices must use ``list()`` in python3
+  - elasticsearch_plugin - ``state`` parameter choices must use ``list()`` in python3
+  - riak - parameters ``wait_for_handoffs`` and ``wait_for_ring`` are ``int`` but the default value was ``False``
```
Comment (felixfontein) [inline changelogs/fragments/1830-valmod_docmissingtype_batch1.yml:4]:
```suggestion
  - riak - parameters ``wait_for_handoffs`` and ``wait_for_ring`` are ``int`` but the default value was ``false``
```

File: changelogs/fragments/1830-valmod_docmissingtype_batch1.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+bugfixes:
+  - kibana_plugin - ``state`` parameter choices must use ``list()`` in python3
```
Comment (felixfontein) [inline changelogs/fragments/1830-valmod_docmissingtype_batch1.yml:2]: Generally, all fragments should look like this:
```suggestion
  - kibana_plugin - ``state`` parameter choices must use ``list()`` in python3 (https://github.com/ansible-collections/community.general/pull/1830).
```

File: changelogs/fragments/1830-valmod_docmissingtype_batch1.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  - elasticsearch_plugin - ``state`` parameter choices must use ``list()`` in python3
+  - riak - parameters ``wait_for_handoffs`` and ``wait_for_ring`` are ``int`` but the default value was ``False``
+  - logstash_plugin - wrapped ``dict.keys()`` with ``list`` for use in ``choices`` setting
+minor_changes:
+  - runit - removed unused code
+  - runit - passing command as ``list`` instead of ``str`` to ``run_command()``
```
Comment (felixfontein) [inline changelogs/fragments/1830-valmod_docmissingtype_batch1.yml:8]:
```suggestion
  - runit - removed unused code, and passing command as ``list`` instead of ``str`` to ``run_command()`` (https://github.com/ansible-collections/community.general/pull/1830).
```
Please move this into the `bugfixes:` section.

File: plugins/modules/files/iso_extract.py [hunk 4]
```diff
@@ -101,7 +101,8 @@ def main():
             image=dict(type='path', required=True, aliases=['path', 'src']),
             dest=dict(type='path', required=True),
             files=dict(type='list', elements='str', required=True),
-            force=dict(type='bool', default=True, aliases=['thirsty']),
+            force=dict(type='bool', default=True, aliases=['thirsty'],
+                       deprecated_aliases=[dict(name='thirsty', version='3.0.0', collection_name='community.general')]),
```
Comment (felixfontein) [inline plugins/modules/files/iso_extract.py:105]: Please remove the below `deprecate()` call. Also, this change does not need a changelog fragment IMO, or only a `bugfixes:` entry with `use proper alias deprecation mechanism for ``thirsty`` alias of ``force``.`

File: plugins/modules/system/runit.py [hunk 5]
```diff
@@ -239,7 +217,7 @@ def main():
             name=dict(type='str', required=True),
             state=dict(type='str', choices=['killed', 'once', 'reloaded', 'restarted', 'started', 'stopped']),
             enabled=dict(type='bool'),
-            dist=dict(type='str', default='runit'),  # @FIXME unused param?
+            dist=dict(type='str', removed_in_version='4.0.0', removed_from_collection='community.general'),
```
Comment (felixfontein) [inline plugins/modules/system/runit.py:220]: Can you put this into a separate PR? This is the only thing we cannot backport to stable-1.

File: changelogs/fragments/1830-valmod_docmissingtype_batch1.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+bugfixes:
+  - kibana_plugin - ``state`` parameter choices must use ``list()`` in python3 (https://github.com/ansible-collections/community.general/pull/1830).
+  - elasticsearch_plugin - ``state`` parameter choices must use ``list()`` in python3 (https://github.com/ansible-collections/community.general/pull/1830).
+  - riak - parameters ``wait_for_handoffs`` and ``wait_for_ring`` are ``int`` but the default value was ``false`` (https://github.com/ansible-collections/community.general/pull/1830).
+  - logstash_plugin - wrapped ``dict.keys()`` with ``list`` for use in ``choices`` setting (https://github.com/ansible-collections/community.general/pull/1830).
+  - iso_extract - use proper alias deprecation mechanism for ``thirsty`` alias of ``force`` (https://github.com/ansible-collections/community.general/pull/1830).
+minor_changes:
+  - runit - removed unused code (https://github.com/ansible-collections/community.general/pull/1830).
```
Comment (felixfontein) [inline changelogs/fragments/1830-valmod_docmissingtype_batch1.yml:8]:
```suggestion
```

Comment (russoz) [reply changelogs/fragments/1830-valmod_docmissingtype_batch1.yml:8]: This code removal is still there. Doesn't that warrant a changelog ?

Comment (felixfontein) [reply changelogs/fragments/1830-valmod_docmissingtype_batch1.yml:8]: That part is already in the changelog fragment in the line below :)

**Comment by russoz** [comment]
.... and rebasing again (would need to trigger tests again).

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for fixing these!

**Comment by felixfontein** [comment]
Manual backport to stable-1 in #1846.
